runtime/runtime.go

```diff
@@ -196,7 +196,14 @@ type ContainerData struct {
 	serviceCtx *narwhal.ServiceContext
 }
 
+// Environment returns a map of enviroment variables with each container IP adress
+//  it also fill up the ip map, that is used in IP
 func (c ContainerData) Environment(ctx context.Context) map[string]string {
+	// As it's called only once
+	if c.ip == nil {
+		c.ip = make(map[string]string)
```

Comment:
I'm not sure I understand the effect of this assignment. From [the spec](https://golang.org/ref/spec#Index_expressions):

> if the map is `nil` or does not contain such an entry, `a[x]` is the zero value for the element type of [the map]

Reply:
I was getting a SIGSEV for trying to assign a key+value to a nil map. But that's why this is a draft, I was working a better way to initialize this

Reply:
That makes sense. In the future, having a new test case in the PR or expanding the description would help me understand the change.